### PR TITLE
Add FAQ for splitting projects

### LINKED ISSUE #397: [question] How to handle installers
While looking at #358, I noticed gmp uses bison and flex which are not (yet) packaged by CCI.

These packages consist of an installer and library component.
Is there a protocol/consensus about how to handle such packages?

### PR DESCRIPTION
**Pull request by uilianries**
closes #397

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
Maybe it could be recommended to use 2 profiles even for native build. Indeed, having the same recipe in requirements and build requirements leads to lots of issues without 2 profiles.

**Comment by uilianries** [comment]
@SpaceIm done

**Review by prince-chrismc — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by mathbunnyru — CHANGES_REQUESTED** [review]
Let's maybe add a line, that says, that `tools` option could be added, to provide binaries, as well as the libraries?

File: docs/faqs.md [hunk 1]
```diff
@@ -161,6 +161,11 @@ and [libcurl](https://github.com/conan-io/conan-center-index/blob/f834ee1c825641
 However, if logic is too complex (this is subjective and depends on the Conan review team) then just remove the option.
 After one month, we will welcome a PR removing the option that was deprecated.
 
+
+## Can I split a project into an installer and library package?
+
+No. Some projects provide more than a simple library, but also applications. For those projects, both libraries and executables should be kept together under the same Conan package. In the past, we tried to separate popular projects, like Protobuf, and it proved to be a complex and hard task to be maintained, requiring custom patches to disable parts of the building. Also, with the [context](https://docs.conan.io/en/latest/systems_cross_building/cross_building.html#conan-v1-24-and-newer) feature, we can use the same package as build requirement, for the same build platform, and as a regular requirement, for the host platform, when cross-building. It's recommended using 2 profiles in that case, one for build platform (where the compilation tools are being executed) and one for host platform (where the generated binaries will run).
```
Comment (mathbunnyru) [inline docs/faqs.md:167]:
```suggestion
No. Some projects provide more than a simple library, but also applications. For those projects, both libraries and executables should be kept together under the same Conan package. In the past, we tried to separate popular projects, like Protobuf, and it proved to be a complex and hard task to be maintained, requiring custom patches to disable parts of the building. Also, with the [context](https://docs.conan.io/en/latest/systems_cross_building/cross_building.html#conan-v1-24-and-newer) feature, we can use the same package as build requirement, for the same build platform, and as a regular requirement, for the host platform, when cross-building. It's recommended using 2 profiles in that case, one for build platform (where the compilation tools are being executed) and one for host platform (where the generated binaries will run). To have both library and applications, option `tools` may be added to the recipe.
```

**Review by mathbunnyru — CHANGES_REQUESTED** [review]

**Review by SSE4 — APPROVED** [review]

File: docs/faqs.md (hunk 1 above)
Comment (uilianries) [reply docs/faqs.md:167]: We don't have a rule about it. Some projects have their own options. We can add it, but I would for a separated PR, with more comments. This PR is dedicated to a more specific topic. Also, if we accept "tools" as a regular option, we will need to add it in Hooks.

Comment (mathbunnyru) [reply docs/faqs.md:167]: Ok, then let's merge this as is.

**Review by mathbunnyru — APPROVED** [review]

**Pull request merged**